usecase/parser/utils/parsermanager.go

```diff
@@ -0,0 +1,77 @@
+package utils
+
+import (
+	"github.com/crypto-com/chain-indexing/entity/command"
+	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+	"github.com/crypto-com/chain-indexing/usecase/event"
+	"github.com/crypto-com/chain-indexing/usecase/model"
+)
+
+type CosmosParserManager struct {
+	store  map[string]CosmosParserBlockHeight
+	logger applogger.Logger
+	config CosmosParserManagerConfig
+}
+
+type CosmosParserBlockHeight map[uint64]CosmosParser
+
+type CosmosParserManagerConfig struct {
+	CosmosVersionBLockHeight
+}
+
+type CosmosVersionBLockHeight struct {
+	V0_43_0 uint64
+}
+
+type CosmosParserManagerParams struct {
+	Logger applogger.Logger
+	Config CosmosParserManagerConfig
+}
+
+type CosmosParser func(
+	CosmosParserParams,
+) []command.Command
+
+type CosmosParserParams struct {
+	MsgCommonParams event.MsgCommonParams
+	TxsResult       model.BlockResultsTxsResult
+	MsgIndex        int
+	Msg             map[string]interface{}
+}
+
+func NewCosmosParserManager(params CosmosParserManagerParams) *CosmosParserManager {
+	cpm := &CosmosParserManager{
+		store:  make(map[string]CosmosParserBlockHeight),
+		logger: params.Logger,
+		config: params.Config,
+	}
+
+	return cpm
+}
+
+func (cpm *CosmosParserManager) GetConfig() CosmosParserManagerConfig {
+	return cpm.config
+}
+
+func (cpm *CosmosParserManager) RegisterParser(name string, fromHeight uint64, parser CosmosParser) {
+	if cpm.store[name] == nil {
+		cpm.store[name] = make(map[uint64]CosmosParser)
+	}
+	cpm.store[name][fromHeight] = parser
+}
+
+func (cpm *CosmosParserManager) GetParser(name string, blockHeight uint64) CosmosParser {
+	parserVersions, ok := cpm.store[name]
+	if !ok {
+		cpm.logger.Errorf("Requesting invalid parser :%s", name)
+	}
+	resultBlockHeight := uint64(0)
+	resultParser := parserVersions[resultBlockHeight]
+	for fromBlockHeight, parser := range parserVersions {
+		if fromBlockHeight <= blockHeight && fromBlockHeight > resultBlockHeight {
+			resultParser = parser
```

Comment:
Here shall we also update the `resultBlockHeight`?

Reply:
Why do we need to track a resultBlockHeight here?

Reply:
since mapping may not be sequential, need it for height comparison

Reply:
Got it, I think this is not very self explanatory actually. Can you see if we can move it to some functions or update the variable naming to reveal the logic intention? Thank you!

Reply:
True, got your point.

Reply:
Updated the naming for better explanation